src/data/set/intervals/unordered_interval.lean

```diff
@@ -60,6 +62,10 @@ interval_of_gt (lt_of_not_ge h)
 lemma interval_of_not_ge (h : ¬ b ≤ a) : [a, b] = Icc a b :=
 interval_of_lt (lt_of_not_ge h)
 
+lemma interval_eq_union : interval a b = Icc a b ∪ Icc b a := by rw [Icc_union_Icc', max_comm]; refl
+
+lemma mem_interval : a ∈ interval b c ↔ b ≤ a ∧ a ≤ c ∨ c ≤ a ∧ a ≤ b := by simp [interval_eq_union]
```

Comment:
Does this let you golf any later proofs? Perhaps `left_mem_interval` might be simpler?